packages/components/src/internal/components/chart/ChartFieldOption.tsx

```diff
@@ -86,6 +86,8 @@ export const ChartFieldOption: FC<ChartFieldOptionProps> = memo(props => {
             parseFloat(scale.max.toString()) <= parseFloat(scale.min.toString()),
         [scale]
     );
+    // Issue 52050: use fieldKey for special characters
+    const selectInputValue = useMemo(() => fieldValue?.data.fieldKey ?? fieldValue?.value, [fieldValue]);
```

Comment:
In chart ChartBuilderModal cautious in a different way: `field.data?.fieldKey`, while here we do `fieldValue?.data.fieldKey`. Is it intentional to be cautious in two different ways?